make/e2e-setup.mk

```diff
@@ -172,13 +185,18 @@ feature_gates_controller := $(subst $(space),\$(comma),$(filter AllAlpha=% AllBe
 feature_gates_webhook := $(subst $(space),\$(comma),$(filter AllAlpha=% AllBeta=% AdditionalCertificateOutputFormats=% LiteralCertificateSubject=%,   $(subst $(comma),$(space),$(FEATURE_GATES))))
 feature_gates_cainjector := $(subst $(space),\$(comma),$(filter AllAlpha=% AllBeta=%, $(subst $(comma),$(space),$(FEATURE_GATES))))
 
+# Install cert-manager with E2E specific images and deployment settings.
+# The values.best-practice.yaml file is applied for compliance with the
+# Kyverno policy which has been installed in a pre-requisite target.
+#
 # TODO: move these commands to separate scripts for readability
 #
 # ⚠ The following components are installed *before* cert-manager:
 # * GatewayAPI: so that cert-manager can watch those CRs.
-# * Kyverno: so that it can check the cert-manager manifests against the policy in `config/kyverno/`.
+# * Kyverno: so that it can check the cert-manager manifests against the policy in `config/kyverno/`
+#		(only installed if E2E_SETUP_OPTION_BESTPRACTICE is set).
 .PHONY: e2e-setup-certmanager
-e2e-setup-certmanager: $(BINDIR)/cert-manager.tgz $(foreach binaryname,controller acmesolver cainjector webhook ctl,$(BINDIR)/containers/cert-manager-$(binaryname)-linux-$(CRI_ARCH).tar) $(foreach binaryname,controller acmesolver cainjector webhook ctl,load-$(BINDIR)/containers/cert-manager-$(binaryname)-linux-$(CRI_ARCH).tar) e2e-setup-gatewayapi e2e-setup-kyverno $(BINDIR)/scratch/kind-exists | $(NEEDS_KUBECTL) $(NEEDS_KIND) $(NEEDS_HELM)
+e2e-setup-certmanager: $(BINDIR)/cert-manager.tgz $(foreach binaryname,controller acmesolver cainjector webhook ctl,$(BINDIR)/containers/cert-manager-$(binaryname)-linux-$(CRI_ARCH).tar) $(foreach binaryname,controller acmesolver cainjector webhook ctl,load-$(BINDIR)/containers/cert-manager-$(binaryname)-linux-$(CRI_ARCH).tar) e2e-setup-gatewayapi $(if $(E2E_SETUP_OPTION_BESTPRACTICE),e2e-setup-kyverno) $(BINDIR)/scratch/kind-exists | $(NEEDS_KUBECTL) $(NEEDS_KIND) $(NEEDS_HELM)
```

Comment:
suggestion: We could avoid always having a network call here when setting up a new cluster, by caching the file locally if we move away from using `master` as a ref in `E2E_SETUP_OPTION_BESTPRACTICE_HELM_VALUES_URL`.

Doesn't need to block and I've not tested the below, but it's maybe worth considering?

```suggestion
E2E_SETUP_OPTION_REF ?= f0cc0f3b88846969dd7e9894cddd43391a3135d1
E2E_SETUP_OPTION_BESTPRACTICE_HELM_VALUES_URL ?= https://raw.githubusercontent.com/cert-manager/website/$(E2E_SETUP_OPTION_REF)/public/docs/installation/best-practice/values.best-practice.yaml

E2E_SETUP_OPTION_DEPENDENCIES ?= $if(E2E_SETUP_OPTION_BESTPRACTICE,e2e-setup-kyverno $(BINDIR)/scratch/values-bestpractice-$(E2E_SETUP_OPTION_REF).yaml)

$(BINDIR)/scratch/values-bestpractice-$(E2E_SETUP_OPTION_REF).yaml: | $(BINDIR)/scratch
	$(CURL) $(E2E_SETUP_OPTION_BESTPRACTICE_HELM_VALUES_URL) -o $@


e2e-setup-certmanager: $(BINDIR)/cert-manager.tgz $(foreach binaryname,controller acmesolver cainjector webhook ctl,$(BINDIR)/containers/cert-manager-$(binaryname)-linux-$(CRI_ARCH).tar) $(foreach binaryname,controller acmesolver cainjector webhook ctl,load-$(BINDIR)/containers/cert-manager-$(binaryname)-linux-$(CRI_ARCH).tar) e2e-setup-gatewayapi $(E2E_SETUP_OPTION_DEPENDENCIES) $(BINDIR)/scratch/kind-exists | $(NEEDS_KUBECTL) $(NEEDS_KIND) $(NEEDS_HELM)

...
		$(if $(E2E_SETUP_OPTION_BESTPRACTICE),--values=$(BINDIR)/scratch/values-bestpractice-$(E2E_SETUP_OPTION_REF).yaml \
```